src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -291,6 +352,37 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(PhysicalLocation v2Ph
             return physicalLocation;
         }
 
+        internal PhysicalLocationVersionOne CreatePhysicalLocation(FileLocation v2FileLocation)
+        {
+            PhysicalLocationVersionOne physicalLocation = null;
+
+            if (v2FileLocation != null)
+            {
+                physicalLocation = new PhysicalLocationVersionOne
+                {
+                    Uri = v2FileLocation.Uri,
+                    UriBaseId = v2FileLocation.UriBaseId
+                };
+            }
+
+            return physicalLocation;
+        }
+
+        internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+        {
```

Comment:
This can't be right. A v1 `physicalLocation` requires a [`uri` property](https://rawgit.com/sarif-standard/sarif-spec/master/Static%20Analysis%20Results%20Interchange%20Format%20(SARIF).html#physicallocation-uri). You'll see that there were a couple of exceptions to this in the v1 spec, but neither of those exceptions can be represented in v2, so the result is, when converting from v2 to v1, you always need a `uri`.

I haven't yet see the call to this method, so I don't yet know how to avoid this... stay tuned.

Reply:
**UPDATE* I was wrong. The place you use this is indeed one of the exceptions in the v1 spec. This is fine.